src/aiida_quantumespresso/workflows/ph/base.py

```diff
@@ -257,13 +258,39 @@ def handle_out_of_walltime(self, node):
     @process_handler(priority=410, exit_codes=PhCalculation.exit_codes.ERROR_CONVERGENCE_NOT_REACHED)
     def handle_convergence_not_reached(self, node):
         """Handle `ERROR_CONVERGENCE_NOT_REACHED` exit code: decrease the mixing beta and restart."""
+        self.ctx.restart_calc = node
         factor = self.defaults.delta_factor_alpha_mix
+
+        nmix_ph = self.ctx.inputs.parameters.get('INPUTPH', {}).get('nmix_ph', self.defaults.nmix_ph)
+
+        if nmix_ph < 20:  # in all these cases, better trying to use more iterations in the mixing first
+            if nmix_ph < 8:  # 8~20 is the recommended range on the Quantum ESPRESSO documentation
+                nmix_ph_new = 8
+            if nmix_ph >= 8:
+                nmix_ph_new = 20
+
+            self.ctx.inputs.parameters.setdefault('INPUTPH', {})['nmix_ph'] = nmix_ph_new
+
+            action = f'increased number of mixing iteration from {nmix_ph} to {nmix_ph_new} and restarting'
+            self.report_error_handled(node, action)
+
+            return ProcessHandlerReport(True)
+
+        # now try playing with the mixing parameters
```

Comment:
In my runs I had immediately set `alpha_mix` to 0.4 in the `overrides`. I didn't gather any statistics, but I did get a noticable improvement in convergence. Something to consider?

Reply:
One additional note on the line below this one (which was already there ^^): The user can also set `alpha_mix` instead of `alpha_mix(1)` and this would work fine. If we only look for `alpha_mix(1)` and then set it, I think we override the `alpha_mix` specified by the user.

Reply:
I've always just used `alpha_mix`:

```
mabercx@uan04:/pfs/lustrep3/scratch/project_465000106/mabercx/88/a0/a0e0-3391-4e6b-b4da-fc98886eaa4a> grep alpha_mix aiida.in
  alpha_mix =   4.0000000000d-01
mabercx@uan04:/pfs/lustrep3/scratch/project_465000106/mabercx/88/a0/a0e0-3391-4e6b-b4da-fc98886eaa4a> grep alpha_mix aiida.out | head -n 5
      thresh= 1.000E-02 alpha_mix =  0.400 |ddv_scf|^2 =  1.410E-08
      thresh= 1.187E-05 alpha_mix =  0.400 |ddv_scf|^2 =  3.525E-09
      thresh= 5.937E-06 alpha_mix =  0.400 |ddv_scf|^2 =  7.637E-11
      thresh= 8.739E-07 alpha_mix =  0.400 |ddv_scf|^2 =  2.221E-13
      thresh= 4.713E-08 alpha_mix =  0.400 |ddv_scf|^2 =  3.612E-14
```